### PR TITLE
Nc news11

### PR DESCRIPTION
**Pull request by amrutha-chil**

### CODE REVIEW

File: __tests__/nc-news5.test.js [hunk 1]
```diff
@@ -67,4 +67,41 @@ describe("/api/articles", () => {
...
+        const { articles } = body;
+        articles.forEach((article) => {
+          expect(article).toHaveProperty("topic", "mitch");
+        });
+      });
+  });
+  test("should return  if an invalid query is made", () => {
+    return request(app)
+      .get("/api/articles?topic=xylophone")
+      .expect(200)
```
Comment (NRMSMITH) [inline __tests__/nc-news5.test.js]: This should really be a 404 - the topic doesn't exist so therefore cannot be found

File: __tests__/nc-news5.test.js [hunk 2]
```diff
@@ -67,4 +67,41 @@ describe("/api/articles", () => {
...
+      });
+  });
+  test("should sort the articles by title in ascending order", () => {
+    return request(app)
+      .get("/api/articles?sort_by=invalidQuery&order=asc")
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.message).toBe("Bad request");
+      });
+  });
```
Comment (NRMSMITH) [inline __tests__/nc-news5.test.js:104]: A few more here: 
-200: topic=paper exists
-400: order=bananas

**Review by NRMSMITH — CHANGES_REQUESTED** [review]
A few more tests to implement!

File: __tests__/nc-news5.test.js (hunk 1 above)
Comment (amrutha-chil) [reply __tests__/nc-news5.test.js]: I have tried to implement this, but as the api is correct it is not giving me an error but only an empty array. Which is why I am checking the length with 200 code status.

File: __tests__/nc-news5.test.js (hunk 2 above)
Comment (amrutha-chil) [reply __tests__/nc-news5.test.js:104]: completed

File: models/nc-news.model.js [hunk 3]
```diff
@@ -43,9 +63,11 @@ const getArticlesCommentsCount = () => {
   articles.article_img_url,
   COUNT(comments) AS comment_count
   FROM articles JOIN comments ON articles.article_id = comments.article_id 
-  GROUP BY articles.article_id, articles.created_at
-  ORDER BY articles.created_at DESC;
-  `
+  ${topic ? "WHERE topic = $1" : ""}
```
Comment (hevdevs) [inline models/nc-news.model.js]: We need to do some kind of check to see if the provided topic is valid as right now it doesn't look like this is being checked anywhere?

Have another look at the notes on complex queries 👀 https://notes.northcoders.com/courses/js-back-end/complex-queries

**Review by hevdevs — CHANGES_REQUESTED** [review]
Hi Amrutha! Just the one refactor suggestion to look at in your model to get that test for invalid topic to throw a 404 instead of a 200 as mentioned by Niamh earlier

**Comment by Dutton1990** [comment]
Thanks for making those changes Amrutha! You can merge this one, great work!

**Pull request merged**